src/wristpy/core/orchestrator.py

```diff
@@ -0,0 +1,259 @@
+            reference_measure.time <= window.wakeup
+        )
+        sleep_array[sleep_mask] = 1
+
+    return sleep_array
+
+
+def format_nonwear_data(
+    nonwear_data: models.Measurement, reference_measure: models.Measurement
+) -> np.ndarray:
+    """Format nonwear data into an array for saving.
+
+    Args:
+        nonwear_data: The nonwear measurement.
+        reference_measure: The measure from which the temporal resolution will be taken.
+            Any epoch1 measure can be used, but enmo is used for consistency.
+
+    Returns:
+        1-D binary np.ndarray, with 1 indicating nonwear. Will be of the same length as
+            the timestamps in the reference_measure.
+
+    """
+    nonwear_df = pl.DataFrame(
+        {
+            "nonwear": nonwear_data.measurements.astype(np.int64),
+            "time": nonwear_data.time,
+        }
+    ).set_sorted("time")
+
+    nonwear_upsample = nonwear_df.upsample(time_column="time", every="5s").fill_null(
+        strategy="forward"
+    )
+
+    end_sequence = pl.repeat(
+        nonwear_upsample["nonwear"][-1],
+        n=(len(reference_measure.time) - len(nonwear_upsample["time"])),
+        dtype=pl.Int64,
+        eager=True,
+    )
+    full_nonwear_array = pl.concat([nonwear_upsample["nonwear"], end_sequence])
+
+    return full_nonwear_array.to_numpy()
+
+
+def run(
+    input: pathlib.Path,
+    output: Optional[pathlib.Path] = None,
+    settings: config.Settings = config.Settings(),
+    calibrator: calibration.Calibration = calibration.Calibration(),
+    detect_nonwear_kwargs: Optional[Dict[str, Any]] = None,
+) -> Results:
+    """Runs wristpy.
+
+    Args:
+        input: Path to the input file to be read. Currently supports .bin and .gt3x
+        output: Path to save data to. The path should end in the file name to be given
+            to the save data. Two files will be saved, each with the given file name and
+            the _raw_time or _epoch1 label after. Currently supports saving in .csv and
+            .parquet
+        settings: The settings object from which physical activity levels are taken.
+        calibrator: The calibrator to be used on the input data.
+        detect_nonwear_kwargs: The arguments to the nonwear function delivered as a
+            dictionary. Arguements are short_epoch_length, n_short_epoch_in_long_epoch,
+            std_criteria, range_criteria.
+
+    Returns:
+        All calculated data in a save ready format as a Results object.
+
+    """
+    watch_data = readers.read_watch_data(input)
+    try:
+        calibrated_acceleration = calibrator.run(watch_data.acceleration)
+    except Exception as e:
+        print("Calibration FAILED:", str(e))
+        print("Proceeding without calibration.")
+        calibrated_acceleration = watch_data.acceleration
+    enmo = metrics.euclidean_norm_minus_one(calibrated_acceleration)
+    enmo_epoch1 = computations.moving_mean(enmo)
```

Comment:
So I know there's a default on moving_mean, but now that I read the usage here I'd be inclined to remove the default entirely and enforce an explicit window everywhere so it's always clear what the epoch is. @Asanto32 thoughts?

Reply:
Either way, removing the default would be outside this PR.

Reply:
Sorry I missed this. But we can remove the default, I think I was setting it up that way since the 5s "epoch_1" timing was so commonly used in a lot of the GGIR functionality